contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol

```diff
@@ -388,18 +370,19 @@ abstract contract FunctionsBilling is HasRouter, IFunctionsBilling {
     // (1e18 juels/link) * (wei/gas) / (wei/link) = juels per gas
     uint256 juelsPerGas = (1e18 * tx.gasprice) / uint256(weiPerUnitLink);
     // Gas overhead without callback
-    uint96 gasOverheadJuels = uint96(juelsPerGas * commitment.gasOverhead);
-    uint96 costWithoutFulfillment = gasOverheadJuels + commitment.donFee;
+    uint96 gasOverheadJuels = uint96(
+      juelsPerGas * (commitment.gasOverheadBeforeCallback + commitment.gasOverheadAfterCallback)
```

Comment:
Why is this changing? Some unrelated bugfix?

Reply:
reducing one variable (costWithoutFulfillment) to help with callstack size
logic is the same

Reply:
previously, we just had a gasOverhead and now we have a gasOverheadAfterCallback and gasOverheadBeforeCallback. @justinkaseman correct me if I am wrong, but was this done just to make it easier to reason through the two places where we need to account for gas overhead?
ie: is the reason for splitting gasOverhad into 2 variables just for convenience?

Reply:
Actually, I think I see the reason now.  Previously, we had 2 different Commitment structures
This was the one in FunctionsBilling
```
  struct Commitment {
    uint64 subscriptionId;
    address client;
    uint32 callbackGasLimit;
    address don;
    uint96 adminFee;
    uint96 estimatedTotalCostJuels;
    uint80 donFee;
    uint32 timestamp;
    uint40 gasOverhead;
    uint256 expectedGasPrice;
  }
```
And this was the on in FunctionsSubscriptions
```
  struct Commitment {
    uint96 adminFee;
    address coordinator;
    address client;
    uint64 subscriptionId;
    uint32 callbackGasLimit;
    uint96 estimatedCost;
    uint40 timeoutTimestamp;
    uint120 gasAfterPaymentCalculation;
  }
```

Now we have a unified Commitment structure in IFunctionsRequest
```
  struct Commitment {
    uint96 adminFee;
    address coordinator;
    address client;
    uint64 subscriptionId;
    uint32 callbackGasLimit;
    uint96 estimatedTotalCostJuels;
    uint40 timeoutTimestamp;
    bytes32 requestId;
    uint80 donFee;
    uint40 gasOverheadBeforeCallback; // old FunctionsBilling.Commitment.gasOverhead === new gasOverheadBeforeCallback + gasOverheadAfterCallback
    uint40 gasOverheadAfterCallback; // old FunctionsSubscriptions.Commitment.gasAfterPaymentCalculation === new gasOverhadAfterCallback
  }
}
```

Is this correct @justinkaseman

Reply:
Yeah you have it right. It's just an effect from having a standard interface